### ISSUE TITLE
AWS SNS delivery failure feedback not set

### ISSUE THREAD

**Issue by grimm26** [terraform-providers/terraform-provider-aws#3354]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Terraform v0.11.3

### Affected Resource(s)
- aws_sns_topic

### Terraform Configuration Files
```hcl
provider "aws" {}

resource "aws_sns_topic" "m" {
  name                                = "mktest2"
  lambda_success_feedback_role_arn    = "arn:aws:iam::112233445566:role/SNSSuccessFeedback"
  lambda_failure_feedback_role_arn    = "arn:aws:iam::112233445566:role/SNSFailureFeedback"
  lambda_success_feedback_sample_rate = "100"
}
```

### Debug Output
https://gist.github.com/grimm26/3b88f0720cc68996959e8f739d075dda

### Expected Behavior
Create success and failure feedback role arn settings in the topic.

### Actual Behavior
only the success feedback arn was created.

### Steps to Reproduce
See gist.  Just try to use the parameters for lambda_success_feedback_role_arn and lambda_failure_feedback_role_arn

### References
https://github.com/terraform-providers/terraform-provider-aws/pull/2872

**Comment by grimm26** [comment]
actually, I'm having it sometimes do nothing or other pieces alone.
same config.
```hcl
> $ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + aws_sns_topic.m
      id:                                  <computed>
      arn:                                 <computed>
      lambda_failure_feedback_role_arn:    "arn:aws:iam::643927032162:role/SNSFailureFeedback"
      lambda_success_feedback_role_arn:    "arn:aws:iam::643927032162:role/SNSSuccessFeedback"
      lambda_success_feedback_sample_rate: "100"
      name:                                "mktest2"
      policy:                              <computed>


Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_sns_topic.m: Creating...
  arn:                                 "" => "<computed>"
  lambda_failure_feedback_role_arn:    "" => "arn:aws:iam::643927032162:role/SNSFailureFeedback"
  lambda_success_feedback_role_arn:    "" => "arn:aws:iam::643927032162:role/SNSSuccessFeedback"
  lambda_success_feedback_sample_rate: "" => "100"
  name:                                "" => "mktest2"
  policy:                              "" => "<computed>"
aws_sns_topic.m: Creation complete after 0s (ID: arn:aws:sns:us-east-2:643927032162:mktest2)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

> $ aws sns get-topic-attributes --topic-arn arn:aws:sns:us-east-2:643927032162:mktest2 --query 'Attributes.[LambdaSuccessFeedbackRoleArn, LambdaFailureFeedbackRoleArn,LambdaSuccessFeedbackSampleRate]'
[
    null,
    null,
    "100"
]
```
debug output: https://gist.github.com/grimm26/d0e9097b4e8b097a6553655bdefb57cd

**Comment by bflad** [comment]
@grimm26 can you confirm this works with multiple applies (one for each attribute)? I think I see the code error, but I'm really amazed the existing acceptance test didn't catch this 😓 .

**Comment by grimm26** [comment]
@bflad yep, if I comment the feedback setting out, apply, and then uncomment them one by one and apply for each, they all get applied.

**Comment by bflad** [comment]
Submitted bugfix PR: #3360

**Comment by bflad** [comment]
The fix for this has been merged into master and will be released with v1.10.0 of the AWS provider, likely at the end of this week.

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
